discretisedfield/field.py

```diff
@@ -3181,957 +3180,173 @@ def _fromhdf5(cls, filename):
-
-        figsize : (2,) tuple, optional
-
-            The size of a created figure if ``ax`` is not passed. Defaults to
-            ``None``.
-
-        filter_field : discretisedfield.Field, optional
-
-            A scalar field used for determining whether certain discretisation
-            cells are coloured. More precisely, only those discretisation cells
-            where ``filter_field != 0`` are plotted. Defaults to ``None``.
-
-        lightness_field : discretisedfield.Field, optional
-
-            A scalar field used for adding lightness to the color. Field values
-            are hue. Defaults to ``None``.
-
-        colorbar : bool, optional
-
-            If ``True``, colorbar is shown and it is hidden when ``False``.
-            Defaults to ``True``.
-
-        colorbar_label : str, optional
-
-            Colorbar label. Defaults to ``None``.
-
-        multiplier : numbers.Real, optional
-
-            ``multiplier`` can be passed as :math:`10^{n}`, where :math:`n` is
-            a multiple of 3 (..., -6, -3, 0, 3, 6,...). According to that
-            value, the axes will be scaled and appropriate units shown. For
-            instance, if ``multiplier=1e-9`` is passed, the mesh points will be
-            divided by :math:`1\\,\\text{nm}` and :math:`\\text{nm}` units will
-            be used as axis labels. Defaults to ``None``.
-
-        filename : str, optional
-
-            If filename is passed, the plot is saved. Defaults to ``None``.
+    @property
+    def mpl(self):
+        return dfp.Mpl(self)
 
-        Raises
-        ------
-        ValueError
+    @property
+    def k3d(self):
+        return dfp.K3d(self)
 
-            If the field has not been sliced, its dimension is not 1, or the
-            dimension of ``filter_field`` is not 1.
+    # TODO: Should we make this a property?
+    def fft2(self):
+        """2D FFT.
 
-        Example
+        Returns
         -------
-        1. Visualising the scalar field using ``matplotlib``.
-
-        >>> import discretisedfield as df
-        ...
-        >>> p1 = (0, 0, 0)
-        >>> p2 = (100, 100, 100)
-        >>> n = (10, 10, 10)
-        >>> mesh = df.Mesh(p1=p1, p2=p2, n=n)
-        >>> field = df.Field(mesh, dim=1, value=2)
-        ...
-        >>> field.plane('y').mpl_scalar()
-
-        .. seealso:: :py:func:`~discretisedfield.Field.mpl_vector`
+        discretisedfield.Field
 
         """
-        if not hasattr(self.mesh, 'info'):
-            msg = 'The field must be sliced before it can be plotted.'
+        if not self.mesh.attributes['isplane']:
+            msg = ('The field must be sliced before the Fourier transform can'
+                   'be computed in two dimensions.')
             raise ValueError(msg)
-
-        if self.dim != 1:
-            msg = f'Cannot plot dim={self.dim} field.'
+        
+        # TODO: Is it ever possible to have n.count(1) > 1 if filed is not sliced?
+        if self.mesh.n.count(1) > 1:
```

Comment:
The problem here are fields that have been sliced multiple times, e.g. field.plane('x').plane('y').

Reply:
Thanks. In that case, do we also consider that field to be sliced?

Reply:
Checks for slicing is in line 3200, right above.